Replace hardcoded ID column by dynamic one

NobodysNightmare opened the pull request:
## OpenProject work package
- fixes https://community.openproject.org/work_packages/19305
- apparently also implements https://community.openproject.org/work_packages/13960
## Description

The ID column of the WP list was always hardcoded, but still it is possible to add an ID column manually.

I did some very minor changes to allow the dynamic ID column to behave (nearly) exactly the same as the hard coded one and deleted the latter. The only difference I noticed is that the column title changed from `#` to `ID`.

I also included the ID column as one of the default columns according to the settings.
## Remaining Question

Should I also add a migration that adds the ID column to the setting `work_package_list_default_columns`?
The setting already existed and the ID column was already selectable, but apparently the setting itself was ignored...

I'd opt for no, because the setting is older than the work package view and thus **not hiding** the ID column was apparently a bug.

NobodysNightmare committed:
make dynamic ID column behave like hardcoded one

NobodysNightmare committed:
remove hardcoded ID column

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://nobodysnightmare-feature-hardcoded-id-column.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://nobodysnightmare-feature-hardcoded-id-column.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

floriank commented:
@NobodysNightmare Tests fail

NobodysNightmare commented:
I am not surprised by that ^^
Will take care...

myabc commented:
@NobodysNightmare easy-peasy fixes.

NobodysNightmare committed:
change test expectations to "ID" rather than "#"

"#" only makes for a good caption as the header of the table.
In all other places (e.g. the column selection) it sucks to see "#" instead of ID.

Also ID is not much longer than "#", so we do not lose horizontal screen space or something...

NobodysNightmare commented:
I changed the expectations rather than the actual behaviour.

"#" is not a good label in any place, e.g. it sucks when selecting a column. If that should backfire for someone, redirect the fire to me please ^^

myabc commented:
:+1:

myabc commented:
@NobodysNightmare unfortunately Protractor tests are now failing.

NobodysNightmare committed:
update frontend mock to include ID column
